### PR TITLE
Add created_by info to cosmosDB's trace

### PR DESCRIPTION
**Pull request by huaiyan**
# Description

For portal trace, we use one link to show all flow test's traces from same workspace.
Customer must need some way to filter between themselves and others.
So, we add `created_by` here to enable UI add corresponding feature later.

Example:
image

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by riddlexu — APPROVED** [review]

File: src/promptflow/promptflow/azure/_storage/cosmosdb/span.py [hunk 1]
```diff
@@ -23,8 +23,9 @@ class Span:
     resource: dict = None
     id: str = None
     partition_key: str = None
+    created_by: dict = None
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/azure/_storage/cosmosdb/span.py:26]: in which case user will query with tenant_id?

File: src/promptflow/promptflow/_sdk/_service/app.py [hunk 2]
```diff
@@ -74,6 +77,28 @@ def create_app():
...
 
+        def initialize_created_by_info():
+            from promptflow._sdk._configuration import Configuration
+            from promptflow._sdk._utils import extract_workspace_triad_from_trace_provider
+
+            trace_provider = Configuration.get_instance().get_trace_provider()
+            if trace_provider is None or extract_workspace_triad_from_trace_provider(trace_provider) is None:
+                return
+            try:
+                import jwt
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_sdk/_service/app.py:88]: for get token, consider `promptflow.azure._utils.gerneral.get_token?`

File: src/promptflow/promptflow/azure/_storage/cosmosdb/span.py (hunk 1 above)
Comment (huaiyan) [reply src/promptflow/promptflow/azure/_storage/cosmosdb/span.py:26]: Just a common practice for `created_by` info. Don't have very special reason for it.

File: src/promptflow/promptflow/_sdk/_service/app.py (hunk 2 above)
Comment (huaiyan) [reply src/promptflow/promptflow/_sdk/_service/app.py:88]: Fixed. It will be better if could return `decoded_token` instead...

**Review by riddlexu — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Review by PeiwenGaoMS — APPROVED** [review]

**Pull request merged**